### DISCUSSION TITLE
DetailsList - Drag and Drop component

### DISCUSSION THREAD

**Discussion by leahmsft** [microsoft/fluentui#23791] [Q&A] [answered]
Hi, 

Are there any accessibility guidelines for [this ](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/draganddrop)component? What happens if a user can't drag and drop? Are there keyboard controls that can be used?

**Comment by spmonahan** [comment, marked as answer]
Hi @leahmsft,

There are keyboard controls for drag and drop. You can see them in the [keyboard column reorder& resize example](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/keyboardcolumnreorderresize).

I recently [merged a PR that adds keyboard documentation](https://github.com/microsoft/fluentui/pull/23859).

Are there additional accessibility guidelines you need?

**Reply by leahmsft** [reply]
Thanks @spmonahan, are there more specific instructions on how to actually perform the drag and drop (rearrangement of row items) using only the keyboard controls?

**Reply by spmonahan** [reply]
There aren't currently. Would it be helpful to have them as part of the example?

To access the keyboard controls:
1. Tab into the DetailsList to focus the header
2. Use the right and left arrow keys to navigate between column headers
3. Use `space` or `enter` on a column header to open the reorder/resize menu
4. Use the down and up arrow keys to navigation the menu
5. Use `space` or `enter` to activate a menu item

**Reply by leahmsft** [reply]
It seems like this doesn't work for drag and drop. If there isn't any way to drag and drop using keyboard controls, then this component isn't accessibility friendly. Are there any plans to improve this? Are there any other alternative components that we can use instead?

**Reply by spmonahan** [reply]
Can you provide a detailed description of the feature you're asking about? I want to make sure we're talking about the same thing.

**Reply by leahmsft** [reply]
I'm curious about the [drag and drop](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/draganddrop) capability of the DetailsList. I can rearrange the row items by clicking and holding with a mouse, and then dragging it into a new spot. For example, This is what my list looks like initially:
<img width="665" alt="image" src="https://user-images.githubusercontent.com/101210285/178545793-334e8196-f041-4a18-87e3-ada51c710f9d.png">
Then I can drag/drop using a mouse and rearranged the items in the list (item 0 moved below item 2 now):
<img width="686" alt="image" src="https://user-images.githubusercontent.com/101210285/178546029-939a3f5c-320d-440d-bef1-d4bccf3b4ceb.png">

How can I achieve this without a mouse and using only keyboard controls?

**Reply by spmonahan** [reply]
Dragging and dropping rows via keyboard is not currently supported. Only columns can be reordered via keyboard. I've [opened an issue](https://github.com/microsoft/fluentui/issues/23881) to track the work to add this feature.

Do you have any examples of how you expect this to work? If so, could you add them to the issue please?

**Reply by leahmsft** [reply]
Thanks for tracking this!

**Comment by thieng101** [comment]
Hello, I want to ask does the FluentDataGrid has the drag and drop ability ?
